media/whip_server.go

```diff
@@ -108,7 +110,7 @@ func (s *WHIPServer) CreateWHIP(ctx context.Context, ssr *SwitchableSegmentReade
 		if connectionState == webrtc.ICEConnectionStateFailed {
 			mediaState.CloseError(errors.New("ICE connection state failed"))
 		} else if connectionState == webrtc.ICEConnectionStateClosed {
-			// Business logic when PeerConnection done
+			mediaState.CloseError(ICEDisconnect)
```

Comment:
If the gateway itself closes the connection will we get to this state as well? We might wanna differentiate the cases where the user happily closes it or the server does it for some reason (like errors)